### ISSUE TITLE
multiprocessing issue: AttributeError: '__main__' object has no attribute 'f'

### ISSUE THREAD

**Issue by jondy** [dashingsoft/pyarmor#48]
Obfuscate the following script and run it:
```python
from multiprocessing import Process
import os

def info(title):
    print(title)
    print('module name:', __name__)
    if hasattr(os, 'getppid'):  # only available on Unix
        print('parent process:', os.getppid())
    print('process id:', os.getpid())

def f(name):
    info('function f')
    print('hello', name)

if __name__ == '__main__':
    info('main line')
    p = Process(target=f, args=('bob',))
    p.start()
    p.join()
```
It  reports error:
```
...
AttributeError: Can't get attribute 'f' on <module '__main__' (built-in)>
...
```

**Comment by jondy** [comment]
Because `__spec__` of module `__main__` is not None in obfuscated script, so `multiprocessing` failed. 
Refer to
https://docs.python.org/3.6/reference/import.html?highlight=__spec__#main-spec

There are 2 solutions to solve this problem:

1. Run script with option `-m'

    `python -m mp`

2. Insert a new line before `p.start()`

    __spec__ = None

Or add a extra startup script `main.py`, then import the functions which used  module `multiprocessing`

**Comment by jondy** [comment]
This problem has been fixed in v5.0.1
